mmv1/templates/terraform/pre_update/notebooks_runtime.go.erb

```diff
@@ -0,0 +1,66 @@
+// remove virtualMachine from updateMask
+callSwitch := false
+for i, field := range updateMask {
+    if field == "virtualMachine" {
+      callSwitch = true
+        updateMask = append(updateMask[:i], updateMask[i+1:]...)
+        break
+    }
+}
+
+if callSwitch {
+  // reconstruct url
+  url, err = tpgresource.ReplaceVars(d, config, "{{NotebooksBasePath}}projects/{{project}}/locations/{{location}}/runtimes/{{name}}")
+  if err != nil {
+      return err
+  }
+  url, err = transport_tpg.AddQueryParams(url, map[string]string{"updateMask": strings.Join(updateMask, ",")})
+  if err != nil {
+      return err
+  }
+
+  state := d.Get("state").(string)
+  if state == "INITIALIZING" {
+    time.Sleep(300 * time.Second)
```

Comment:
This seems fairly imprecise. Does the normal update also suffer from this?

Reply:
Nah, just the switch method since it needs the state in "ACTIVE" or "STOPPED". Not a requirement for the update method.